application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/enforcer/DefaultLicensingSecurityCacheRuleInvalidator.java

```diff
@@ -94,25 +100,46 @@ public void invalidate(XarInstalledExtension extension)
     {
         boolean locked = acquireLock();
         try {
-            for (String namespace : extension.getNamespaces()) {
-                if (namespace.startsWith("wiki:")) {
-                    WikiReference wikiRef = new WikiReference(namespace.substring(5));
-                    for (XarEntry entry : extension.getXarPackage().getEntries()) {
-                        securityCache.remove(
-                            securityReferenceFactory.newEntityReference(
-                                documentReferenceResolver.resolve(entry, wikiRef)
-                            )
-                        );
+            Collection<String> namespaces = extension.getNamespaces();
+            if (namespaces != null) {
+                for (String namespace : namespaces) {
+                    if (namespace.startsWith("wiki:")) {
+                        invalidateForWiki(extension, new WikiReference(namespace.substring(5)));
```

Comment:
Even though this is not necessarily part of the issue addressed, it would be nice to replace that magic string and magic number with a string constant (e.g. `WIKI_NAMESPACE`) and a method call to get its length (i.e. `WIKI_NAMESPACE.length()`).